include/deal.II/lac/la_parallel_vector.h

```diff
@@ -376,7 +371,25 @@ namespace LinearAlgebra
        */
       void
       reinit(
-        const std::shared_ptr<const Utilities::MPI::Partitioner> &partitioner);
+        const std::shared_ptr<const Utilities::MPI::Partitioner> &partitioner,
+        const MPI_Comm &comm_sm = MPI_COMM_SELF);
```

Comment:
I realize that we are inconsistent in passing in `MPI_Comm` objects; some we pass by value and others we pass by reference. I would lean towards passing by reference. While we should not clean up everything in this PR, I think it is good to keep an eye on it.

Reply:
Can you add documentation about the optional communicator object?